OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs

```diff
@@ -347,73 +347,31 @@ void CloseWindow()
 			forceStartBin.Get<ButtonWidget>("CANCEL_BUTTON").OnClick = () => panel = PanelType.Players;
 
 			// Options panel
-			var allowCheats = optionsBin.GetOrNull<CheckboxWidget>("ALLOWCHEATS_CHECKBOX");
-			if (allowCheats != null)
-			{
-				var cheatsLocked = new CachedTransform<MapPreview, bool>(
-					map => map.Rules.Actors["player"].TraitInfo<DeveloperModeInfo>().Locked);
-
-				allowCheats.IsChecked = () => orderManager.LobbyInfo.GlobalSettings.AllowCheats;
-				allowCheats.IsDisabled = () => configurationDisabled() || cheatsLocked.Update(Map);
-				allowCheats.OnClick = () => orderManager.IssueOrder(Order.Command(
-						"allowcheats {0}".F(!orderManager.LobbyInfo.GlobalSettings.AllowCheats)));
-			}
-
-			var crates = optionsBin.GetOrNull<CheckboxWidget>("CRATES_CHECKBOX");
-			if (crates != null)
-			{
-				var cratesLocked = new CachedTransform<MapPreview, bool>(map =>
-				{
-					var crateSpawner = map.Rules.Actors["world"].TraitInfoOrDefault<CrateSpawnerInfo>();
-					return crateSpawner == null || crateSpawner.Locked;
-				});
-
-				crates.IsChecked = () => orderManager.LobbyInfo.GlobalSettings.Crates;
-				crates.IsDisabled = () => configurationDisabled() || cratesLocked.Update(Map);
-				crates.OnClick = () => orderManager.IssueOrder(Order.Command(
-					"crates {0}".F(!orderManager.LobbyInfo.GlobalSettings.Crates)));
-			}
-
-			var creeps = optionsBin.GetOrNull<CheckboxWidget>("CREEPS_CHECKBOX");
-			if (creeps != null)
-			{
-				var creepsLocked = new CachedTransform<MapPreview, bool>(map =>
-				{
-					var mapCreeps = map.Rules.Actors["world"].TraitInfoOrDefault<MapCreepsInfo>();
-					return mapCreeps == null || mapCreeps.Locked;
-				});
-
-				creeps.IsChecked = () => orderManager.LobbyInfo.GlobalSettings.Creeps;
-				creeps.IsDisabled = () => configurationDisabled() || creepsLocked.Update(Map);
-				creeps.OnClick = () => orderManager.IssueOrder(Order.Command(
-					"creeps {0}".F(!orderManager.LobbyInfo.GlobalSettings.Creeps)));
-			}
+			var optionCheckboxes = new Dictionary<string, string>()
+			{
+				{ "EXPLORED_MAP_CHECKBOX", "explored" },
+				{ "CRATES_CHECKBOX", "crates" },
+				{ "SHORTGAME_CHECKBOX", "shortgame" },
+				{ "FOG_CHECKBOX", "fog" },
+				{ "ALLYBUILDRADIUS_CHECKBOX", "allybuild" },
+				{ "ALLOWCHEATS_CHECKBOX", "cheats" },
+				{ "CREEPS_CHECKBOX", "creeps" },
+			};
```

Comment:
Doesn't this kind of defeat the purpose of unhardcording the options?

Before reading this PR, my expectation was that each trait that provided a lobby option would contain fields for type (checkbox or dropdown) and precedence (for ordering in the UI), and the UI would then built from template definitions.

Is that going to happen in the future, or is this the final state of the solution?